src/include/rbd/features.h

```diff
@@ -58,14 +61,17 @@
                                          RBD_FEATURE_DATA_POOL      | \
                                          RBD_FEATURE_OPERATIONS     | \
                                          RBD_FEATURE_MIGRATING      | \
-                                         RBD_FEATURE_NON_PRIMARY)
+                                         RBD_FEATURE_NON_PRIMARY    | \
+                                         RBD_FEATURE_DIRTY_CACHE)
 
 /// features that may be dynamically enabled or disabled
 #define RBD_FEATURES_MUTABLE            (RBD_FEATURE_EXCLUSIVE_LOCK | \
                                          RBD_FEATURE_OBJECT_MAP     | \
                                          RBD_FEATURE_FAST_DIFF      | \
                                          RBD_FEATURE_JOURNALING     | \
-                                         RBD_FEATURE_NON_PRIMARY)
+                                         RBD_FEATURE_NON_PRIMARY    | \
+                                         RBD_FEATURE_DIRTY_CACHE)
+
 #define RBD_FEATURES_MUTABLE_INTERNAL   (RBD_FEATURE_NON_PRIMARY)
```

Comment:
Nit: you will need to add it to this setting not `RBD_FEATURES_MUTABLE` since we don't want users to enable/disable this feature via the "rbd feature enable/disable" CLI.

Reply:
The feature needs to add both RBD_FEATURES_MUTABLE_INTERNAL and RBD_FEATURES_MUTABLE.    
RBD_FEATURES_MUTABLE is used in cluster side:
https://github.com/ceph/ceph/blob/master/src/cls/rbd/cls_rbd.cc#L942